src/cppcache/include/geode/DataOutput.hpp

```diff
@@ -378,9 +378,8 @@ class CPPCACHE_EXPORT DataOutput {
      */
   inline void writeFullUTF(const char* value, uint32_t length = 0) {
     if (value != NULL) {
-      int32_t len = getEncodedLength(value, length);
-      uint16_t encodedLen = static_cast<uint16_t>(len > 0xFFFF ? 0xFFFF : len);
```

Comment:
If the query string happened to be longer that 0xFFFF, it would be artificially truncated which often led to errors on the server side.